skycatalogs/utils/sed_tools.py

```
@@ -211,12 +212,107 @@ def create(self, pixel, galaxy_id, redshift_hubble, redshift):
         return seds
 
 
+class TrilegalSedFactory():
+
+    def __init__(self, object_type_config):
+        '''
+        Parameters
+        ----------
+        object_type_config  dict containing section of the sky catalog
+                            config pertaining to object type 'trilegal'
+        '''
+        self._pystellib = None
+
+    def get_sed(self, tri):
+        '''
+        Parameters
+        ----------
+        tri  a trilegal object
+
+        Returns
+        -------
+        galsim sed object.
+
+        If SED file present, find row and wave length values in sed file
+        Otherwise use quantities in main parquet file + pystellibs
+        In either case return unextincted, unnormalized, unthinned SED,
+        just as calculated by pystellib, except converted from Angstroms
+        to nm. The other transformations are handled elsewhere
+        '''
+        if not self._pystellib:
+            from pystellibs import BTSettl
+            self._pystellib = BTSettl(medres=False)
+            self._wl = self._pystellib.wavelength / 10 # convert to nm
+
+        # Get inputs from parquet file
+        native = ['logT', 'logg', 'logL', 'Z']
+        spec_inputs = [tri.get_native_attribute(x) for x in native]
+
+        # May generate a runtime error if parameters are outside
+        # interpolation range
+        try:
+            spectrum = self._pystellib.generate_stellar_spectrum(*spec_inputs)
+        except RuntimeError:
+            return None
+
+        if np.isnan(spectrum[0]):
+            return None
```

Should warnings be emitted in these cases?

---

There are a fair number of sources which fail one of these tests, something like 5%.   Most but not quite all of these are stars with label=9 (one of the columns in the AstroLab table, which corresponds to "post-Asymptotic Giant Branch").   I though about excluding the label 9 stars altogether from the sample and maybe that's the way to go, but for now that seems like too many warnings.

---

I would suggest at least issuing a warning at the first instance of the error, or better, keep a counter of the errors and report the failures later on.  Letting things fail silently 5% of the time doesn't seem like a good policy.